modules/http.lua

```diff
@@ -351,17 +371,20 @@ function ClientRequest:done(data, encoding, cb)
   end
 end
 
-function exports.request(options, onResponse)
+function exports.parse_url(options)
```

Comment:
it is url.parse like node.  this is more of an internal helper for http to simplify string detection